execution/src/posix/execution.cc

```diff
@@ -139,8 +144,37 @@ namespace leatherman { namespace execution {
         return fs.st_mode & S_IXOTH;
     }
 
-    string which(string const& file, vector<string> const& directories)
+    bool is_Builtin(string const& file)
     {
+        string data;
+        bool builtin;
+        try {
+            string cmd = "type ";
+            cmd.append(file);
+            const int max_buffer = 25 + file.length();
+            char buffer[max_buffer];
+            FILE *stream = popen(cmd.c_str(), "r");
+            if (stream) {
+                while (!feof(stream)) {
```

Comment:
You should not use `while(!feof(...))` here.
See https://faq.cprogramming.com/cgi-bin/smartfaq.cgi?id=1043284351&answer=1046476070